addons/resource/static/src/views/fields/many2one_avatar_resource/many2one_avatar_resource_field.js

```diff
@@ -0,0 +1,65 @@
+/** @odoo-module **/
+
+import { registry } from "@web/core/registry";
+import { usePopover } from "@web/core/popover/popover_hook";
+import {
+    Many2OneAvatarUserField,
+    many2OneAvatarUserField,
+} from "@mail/views/web/fields/many2one_avatar_user_field/many2one_avatar_user_field";
+import { AvatarCardResourcePopover } from "@resource/components/avatar_card_resource/avatar_card_resource_popover";
+
+const ExtendMany2OneAvatarToResource = (T) => class extends T {
+    // We choose to extend Many2One_avatar_user instead of patching it as field dependencies need to be added on the widget to manage resources
+    setup() {
+        super.setup(...arguments);
+        if (this.relation === 'resource.resource') {
+            this.avatarCard = usePopover(AvatarCardResourcePopover);
+        }
+    }
+
+    onClickAvatar(ev) {
+        const id = this.props.record.data[this.props.name][0] ?? false;
+        if (id) {
+            if (this.env.isSmall || !this.displayResourcePopover) {
+                return super.onClickAvatar(...arguments);
+            }
+            const target = ev.currentTarget;
+            if (!this.avatarCard.isOpen) {
```

Comment:
Summary our discussion about this: I've copied what was implemented in framework for users for that part. For the many2many an extra check is needed because an avatar card can be open for the current field, but only if it doesn't relate to the same record.